mk/spksrc.cross-env.mk

```diff
@@ -21,12 +21,19 @@ TOOLKIT_ROOT = $(WORK_DIR)/../../../toolkit/syno-$(ARCH)-$(TCVERSION)/work
 ENV += TOOLKIT_ROOT=$(TOOLKIT_ROOT)
 endif
 
+ADDITIONAL_CFLAGS := $(patsubst -O%,,$(ADDITIONAL_CFLAGS))
+ADDITIONAL_CPPFLAGS := $(patsubst -O%,,$(ADDITIONAL_CPPFLAGS))
+ADDITIONAL_CXXFLAGS := $(patsubst -O%,,$(ADDITIONAL_CXXFLAGS))
 ifeq ($(strip $(GCC_DEBUG_INFO)),1)
-GCC_DEBUG_FLAGS = -O0 -g3
-ADDITIONAL_CFLAGS := $(patsubst -O%,,$(ADDITIONAL_CFLAGS)) $(GCC_DEBUG_FLAGS)
-ADDITIONAL_CPPFLAGS := $(patsubst -O%,,$(ADDITIONAL_CPPFLAGS)) $(GCC_DEBUG_FLAGS)
-ADDITIONAL_CXXFLAGS := $(patsubst -O%,,$(ADDITIONAL_CXXFLAGS)) $(GCC_DEBUG_FLAGS)
+  # other options to consider: 
+  #GCC_DEBUG_FLAGS += -fsanitize=address -fsanitize=undefined -fstack-protector-all
+  GCC_DEBUG_FLAGS = -ggdb3 -g3
+else
+  GCC_DEBUG_FLAGS = -O3 -fomit-frame-pointer
 endif
+ADDITIONAL_CFLAGS := $(ADDITIONAL_CFLAGS) $(GCC_DEBUG_FLAGS)
+ADDITIONAL_CPPFLAGS := $(ADDITIONAL_CPPFLAGS) $(GCC_DEBUG_FLAGS)
+ADDITIONAL_CXXFLAGS := $(ADDITIONAL_CXXFLAGS) $(GCC_DEBUG_FLAGS)
```

Comment:
This has a much broader effect than only affecting debug builds.  This is turn will enforce using `-O3 -fomit-frame-pointer` on all buids, which will not work due to some arch specific caveats that requires special flags to be used.  GCC_DEBUG_FLAGS should only be used for GCC_DEBUG_INFO builds and not be affecting other regular builds.

Also I believe this sort of explains the reasons for removing `-O?` in some other Makefiles?